base/applications/sdbinst/sdbinst.c

```diff
@@ -0,0 +1,957 @@
+/*
+ * PROJECT:     ReactOS sdbinst
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Application compatibility database installer
+ * COPYRIGHT:   Copyright 2020 Max Korostil (mrmks04@yandex.ru)
+ */
+
+
+#include <windef.h>
+#include <winbase.h>
+#include <tchar.h>
+#include <winreg.h>
+#include <strsafe.h>
+#include <objbase.h>
+#include <apphelp.h>
+
+
+
+#define APPCOMPAT_CUSTOM_REG_PATH L"Software\\Microsoft\\Windows NT\\CurrentVersion\\AppCompatFlags\\Custom"
+#define APPCOMPAT_LAYERS_REG_PATH L"Software\\Microsoft\\Windows NT\\CurrentVersion\\AppCompatFlags\\Layers"
+#define APPCOMPAT_INSTALLEDSDB_REG_PATH L"Software\\Microsoft\\Windows NT\\CurrentVersion\\AppCompatFlags\\InstalledSDB"
+#define UNINSTALL_REG_PATH L"Software\\Microsoft\\Windows\\CurrentVersion\\Uninstall"
+#define DBPATH_KEY_NAME L"DatabasePath"
+
+HRESULT
+RegisterSdbEntry(
+    _In_ PWCHAR sdbEntryName,
+    _In_ LPCWSTR dbGuid,
+    _In_ ULONGLONG time,
+    _In_ BOOL isInstall,
+    _In_ BOOL isExe)
+{
+    PWCHAR regName;
+    HKEY  hKey = NULL;
+    LSTATUS status;
+    HRESULT hres;
+
+    regName = (PWCHAR)HeapAlloc(GetProcessHeap(), 0, MAX_PATH * sizeof(WCHAR));
```

Comment:
If you know the size of this buffer which remains fixed, why not just use a good old `WCHAR regName[MAX_PATH],`  instead?

Reply:
Yes, in this case i can use WCHAR regName[MAX_PATH]. But if tomorrow I need more memory, for example 1MB - it is too big buffer for stack.

Reply:
yep

Reply:
Yes, but at this point it will simplify code, especially the cleanup.